### PR TITLE
Support additional query params for github authenticator

### PR DESCRIPTION
**Pull request by AnuradhaSK**
## Purpose
Fixes https://github.com/wso2/product-is/issues/11585
Current GitHub authenticator doesn't support passing additional query params. This PR fixes that by adding a new property called `AdditionalQueryParameters`.

imageedit_2_4765366597

### CODE REVIEW

File: component/src/main/java/org/wso2/carbon/identity/authenticator/github/GithubAuthenticator.java [hunk 1]
```diff
@@ -224,11 +228,20 @@ private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientI
         scope.setDisplayOrder(3);
         configProperties.add(scope);
 
+        Property additionalQueryParams = new Property();
+        additionalQueryParams.setName(GithubAuthenticatorConstants.ADDITIONAL_QUERY_PARAMS);
```
Comment (piraveena) [inline component/src/main/java/org/wso2/carbon/identity/authenticator/github/GithubAuthenticator.java:232]: We can use the same constant as OIDCAuthenticator https://github.com/wso2-extensions/identity-outbound-auth-oidc/blob/master/components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java#L275. In that case, we don't need to have a overidden method getQueryString()

**Comment by piraveena** [comment]
Since we have scope and additionalQueryParams in the autheticator behaviour, can you check the behaviour if we provide scope parameter as additionalQueryParams. AFAIK, in OIDCAuthenticator, we give priority to take scope from  queryParam https://github.com/wso2-extensions/identity-outbound-auth-oidc/blob/master/components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java#L377

**Comment by AnuradhaSK** [comment]
Current approach: 

- Since the `scope` property was an initial property, we didn't remove that completely due to support backward compatibility.

- Introduced new property `AdditionalQueryParameters`. The user can define scope param in the same property. (eg:  scope=user repo&login={login_hint}). 
- If the scope property value is defined, that will be considered and the scope param defined in the `AdditionalQueryParameters` will be discarded.
- If  the scope property value is not defined, scope param defined in the `AdditionalQueryParameters` will be considered.

**Comment by piraveena** [comment]
> Current approach:
> 
> * Since the `scope` property was an initial property, we didn't remove that completely due to support backward compatibility.
> * Introduced new property `AdditionalQueryParameters`. The user can define scope param in the same property. (eg:  scope=user repo&login={login_hint}).
> * If the scope property value is defined, that will be considered and the scope param defined in the `AdditionalQueryParameters` will be discarded.
> * If  the scope property value is not defined, scope param defined in the `AdditionalQueryParameters` will be considered.

Please add this to github connector docs

**Review by piraveena — APPROVED** [review]

**Pull request merged**